required/latex-lab/latex-lab-toc-kernel-changes.dtx

```diff
@@ -187,23 +193,29 @@
 % \subsection{\cs{contentsline} extension}
 % [latex-lab]
 % The tagging code must add code at the begin and end of \cs{contentsline}.
-% This code must have access to the arguments. We set it up as config point
-% for now.
+% This code must have access to the arguments. They are passed
+% together in tagging socket.
+% \begin{socketdecl}{tagsupport/toc/contentsline/before,
+%  tagsupport/toc/contentsline/after}
+%  Tagging sockets at the begin and end of contentsline.
+%  They receive \emph{all} contentsline arguments as one argument
```

Comment:
because tagging sockets are currently only defined with 0, 1 and 2 arguments.

Reply:
that's because we thought there will not be more, but if they are needed we should augment the definition rather than establishing a suboptimal interface  without any realy need

Reply:
I didn't find it suboptimal. What is the problem with passing the arguments as brace groups? The argument is specific but that it always the case with sockets. You need to know what it gets.

Reply:
It is not really a convention we use elsewhere

Reply:
What I also see as a downside is that you have to take the argument apart in the plug code to get at the individual ones, either by putting `#1` as the input to a command with 4 arguments or by other means. If you pass 4 args instead you can use the inidivually without any extra processing, no?